types/logger.d.ts

```diff
@@ -0,0 +1,68 @@
+import * as http from 'http'
+import * as net from 'net'
+
+import { FastifyError } from './error'
+import { RawServerBase, RawServerDefault, RawRequestBase, RawRequestDefault, RawReplyBase, RawReplyDefault } from './utils';
+
+/**
+ * Standard Fastify logging function
+ */
+export interface FastifyLogFn {
+  (msg: string, ...args: any[]): void;
+  (obj: object, msg?: string, ...args: any[]): void;
+}
+
+export type LogLevels = 'info' | 'error' | 'debug' | 'fatal' | 'warn' | 'trace'
+
+/**
+ * Fastify Custom Logger options.
+ */
+export interface FastifyLoggerOptions<
+  RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestBase = RawRequestDefault<RawServer>,
+  RawReply extends RawReplyBase = RawReplyDefault<RawServer>
+> {
+  serializers?: {
+    req: (req: RawRequest) => {
+      method: string,
+      url: string,
+      version: string,
+      hostname: string,
+      remoteAddress: string,
+      remotePort: number
+    },
+    err: (err: FastifyError) => {
+      type: string;
+      message: string;
+      stack: string;
+      [key: string]: any;
+    },
+    res: (res: RawReply) => {
+      statusCode: string | number
+    }
+  };
+  info: FastifyLogFn;
+  error: FastifyLogFn;
+  debug: FastifyLogFn;
+  fatal: FastifyLogFn;
+  warn: FastifyLogFn;
+  trace: FastifyLogFn;
+  child: FastifyLogFn | FastifyLoggerOptions<RawServer>;
+  genReqId?: string;
+}
+
+/**
```

Comment:
Should probably move this into TS docs?

Reply:
This is actually on purpose! With VSCode, when you hover over a type it will automatically display the `/** */` comment. See the following screenshot:
<img width="750" alt="Screen Shot 2019-04-05 at 2 21 04 PM" src="https://user-images.githubusercontent.com/16144158/62843727-21cb6f80-bc8a-11e9-9ce6-992b5eb96637.png">

This information should be duplicated in the documentation too thought, good point!

Reply:
And now I realize this is partly useless. If the user is defining their own logging types they'd never be using this object nor the reference. Removing.